core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3397,10 +3396,20 @@ public PhysicalOperation visitTableExecute(TableExecuteNode node, LocalExecution
             return new PhysicalOperation(operatorFactory, outputMapping.buildOrThrow(), context, source);
         }
 
+        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, PlanNode source)
+        {
+            return partitioningScheme
+                    .map(scheme -> getTaskPartitionedWriterCount(session))
+                    .orElseGet(() -> isLocalScaledWriterExchange(source) ? getTaskScaleWritersMaxWriterCount(session) : getTaskWriterCount(session));
+        }
+
         @Override
         public PhysicalOperation visitMergeWriter(MergeWriterNode node, LocalExecutionPlanContext context)
         {
-            context.setDriverInstanceCount(getTaskWriterCount(session));
+            int writerCount = node.getPartitioningScheme()
+                    .map(scheme -> getTaskPartitionedWriterCount(session))
+                    .orElseGet(() -> getTaskWriterCount(session));
```

Comment:
There won't be any `isLocalScaledWriterExchange(source)`?
Can there actually be a merge without partitioning scheme (it's kind of join)?
Is this something we should create an issue for and add `TODO` here?

Reply:
Added a todo